### PR TITLE
Fixed issue: Grid filter does not work with value `0`

### PR DESCRIPTION
**Pull request by jignesh-baldha**
I've fixed UI grid filter does not work with value `0` issue.

### Description
Upon filter with **0** in UI grid was not working, so I've checked for filter does not work with value `0` issue.

And finally found that `Magento\Ui\Component\Filters\Type\Input` class validating filtering data two times by PHP `isset()` and `empty()` function. It looks like `empty()` statement does not require here and `empty()` does return true when we filter with value 0 so our grid filter was not working. I've removed `empty()` condition and it's working fine.

### Fixed Issues (if relevant)
1. magento/magento2#17187: Grid filter does not work with value `0`

### Manual testing scenarios

1. Go to any grid in admin panel (For better understanding go to CMS page list)
2. Insert 0 in any filter
3. Apply filter

It will show all items with no filtering. But we were expecting to Grid is filtered with value `0`. This PR will fix such a filtering issue. 

### Contribution checklist
 - [X] Pull request has a meaningful description of its purpose
 - [X] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @jignesh-baldha. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

**Review by sidolov — CHANGES_REQUESTED** [review]
It would be great if you cover this changes with test, at least unit

File: app/code/Magento/Ui/Component/Filters/Type/Input.php [hunk 1]
```diff
@@ -67,14 +67,12 @@ protected function applyFilter()
         if (isset($this->filterData[$this->getName()])) {
             $value = str_replace(['%', '_'], ['\%', '\_'], $this->filterData[$this->getName()]);
 
-            if (!empty($value)) {
```
Comment (orlangur) [inline app/code/Magento/Ui/Component/Filters/Type/Input.php]: Why not change to comparison with empty string instead?

**Review by orlangur — CHANGES_REQUESTED** [review]

File: app/code/Magento/Ui/Component/Filters/Type/Input.php (hunk 1 above)
Comment (jignesh-baldha) [reply app/code/Magento/Ui/Component/Filters/Type/Input.php]: @orlangur Compared with the empty string.

Comment (orlangur) [reply app/code/Magento/Ui/Component/Filters/Type/Input.php]: @jignesh-baldha thanks! Please change to strict comparison though. And pay attention to Stas suggestion please.

**Comment by jignesh-baldha** [comment]
@sidolov What do you mean, we need to add new unit test for this case? Could you please tell me bit more.

**Comment by sidolov** [comment]
@jignesh-baldha , we already have a unit test for this class - `/app/code/Magento/Ui/Test/Unit/Component/Filters/Type/InputTest.php`
Please, add new dataset with the value '0' for the case that you are trying to fix.

File: app/code/Magento/Ui/Component/Filters/Type/Input.php (hunk 1 above)
Comment (orlangur) [reply app/code/Magento/Ui/Component/Filters/Type/Input.php]: `!== ''`

**Comment by jignesh-baldha** [comment]
@sidolov Added dataset with the value '0'. Could you please check?

**Comment by orlangur** [comment]
@jignesh-baldha I don't see any addition to test.

> return $value === null || $value === '' || (is_array($value) && empty($value));

Could you explain how `0` passed such logic?

**Comment by jignesh-baldha** [comment]
@orlangur I've removed the following function and I've used ` !== ''` instead:

> private function isEmptyValue($value)	
    {	
            return $value === null || $value === '' || (is_array($value) && empty($value));	
    }

And there is already following dataset exist in test.

> [
                'test_date',
                ['test_date' => '0'],
                ['like' => '%0%'],
            ]

**Comment by jignesh-baldha** [comment]
@orlangur Hello, I've to do anything else?

**Comment by orlangur** [comment]
@jignesh-baldha,

> Could you explain how `0` passed such logic?

How did `0` didn't work before? As it is no treated as empty by `isEmptyValue`.

> there is already following dataset exist in test.

Test worked before your changes which means it is insufficient.

**Comment by jignesh-baldha** [comment]
@sidolov I've checked in vanilla Magento. Also, checked unit.
`travis-ci` passed this test and I've improved code, could you please check the code again?

**Comment by sidolov** [comment]
Hi @jignesh-baldha , I don't see any additional dataset for `/app/code/Magento/Ui/Test/Unit/Component/Filters/Type/InputTest.php` in the current changeset

**Comment by jignesh-baldha** [comment]
@sidolov As you said to add new dataset with the value `0` but dataset already exists with the value `0`.

**Comment by sidolov** [comment]
@jignesh-baldha , in that case why we need this fix if the filter with '0' value should work without `isEmptyValue` removing?

**Comment by jignesh-baldha** [comment]
@sidolov https://github.com/magento/magento2/commit/2c3eb76b0341a9b0250182c58719a597fb73637a#diff-eda0c072c5d58e30615600361890a3f5 this changes merged to 2.2-develop branch after I've created PR. You can verify which code is better. If my code is not good then you can close this PR.

Thanks.

**Comment by sidolov** [comment]
@jignesh-baldha I think it would be better to keep `isEmptyValue` method with several checks instead of the only check for empty string. Anyway, thank you for spent time and collaboration!

**Pull request closed without merging**